src/txHashBuilder_test.c

```diff
@@ -225,16 +228,89 @@ static struct {
 	}
 };
 
-static const char* expectedHex = "5B347674B1A9C26E3CB25AC5644FAA7702A43B0187EF91391AE8EC02C214A0A4";
+static const char* expectedHex = "378B60224E51EB566F3BF5FF62EDB18BFCD89B5F3E3B848A7820B26536C4B5D3";
+
+void addPoolRegistrationCertificate(tx_hash_builder_t* builder)
+{
+	pool_registration_params_t poolParams = {};
+
+	size_t poolKeyHashSize = decode_hex(
+	                                 "5631EDE662CFB10FD5FD69B4667101DD289568E12BCF5F64D1C406FC",
+	                                 poolParams.operatorHash, SIZEOF(poolParams.operatorHash)
+	                         );
+	ASSERT(poolKeyHashSize == SIZEOF(poolParams.operatorHash));
+
+	size_t vrfKeyHashSize = decode_hex(
+	                                "198890AD6C92E80FBDAB554DDA02DA9FB49D001BBD96181F3E07F7A6AB0D0640",
+	                                poolParams.vrfKeyHash, SIZEOF(poolParams.vrfKeyHash)
+	                        );
+	ASSERT(vrfKeyHashSize == SIZEOF(poolParams.vrfKeyHash));
+
+	size_t rewardAccountSize = decode_hex(
+	                                   "E03A7F09D3DF4CF66A7399C2B05BFA234D5A29560C311FC5DB4C490711",
+	                                   poolParams.rewardAccount, SIZEOF(poolParams.rewardAccount)
+	                           );
+	ASSERT(rewardAccountSize == SIZEOF(poolParams.rewardAccount));
+
+	poolParams.pledge = 500000000;
+	poolParams.cost = 340000000;
+	poolParams.marginNumerator = 1;
+	poolParams.marginDenominator = 1;
+
+	txHashBuilder_addPoolRegistrationCertificate(
+	        builder,
+	        &poolParams,
+	        1, 3
+	);
+
+	txHashBuilder_addPoolRegistrationCertificate_enterOwners(builder);
+
+	uint8_t owner1[28];
+	size_t owner1Size = decode_hex("3A7F09D3DF4CF66A7399C2B05BFA234D5A29560C311FC5DB4C490711", owner1, SIZEOF(owner1));
+	ASSERT(owner1Size == SIZEOF(owner1));
+
+	txHashBuilder_addPoolRegistrationCertificate_addOwner(builder, owner1, owner1Size);
 
+	txHashBuilder_addPoolRegistrationCertificate_enterRelays(builder);
+
+	uint8_t ipv4[4];
+	size_t ipv4Size = decode_hex("08080808", ipv4, 4);
+	ASSERT(ipv4Size == 4);
+	uint16_t port = 1234;
+
+	txHashBuilder_addPoolRegistrationCertificate_addRelay0(builder, &port, ipv4, ipv4Size, NULL, 0);
+
+	const char* dnsName = "AAAA 2400:cb00:2049:1::a29f:1804";
```

Comment:
for the sake of tests I would pick a legit-looking dns name e.g. `test.test.sth` or something like that , otherwise this has the potential to confuse a casual reader of the code

Reply:
This actually is a legit AAAA DNS record. That `dns_name` is probably a misnomer; CDDL says 
`dns_name ; An A or AAAA DNS record`, so it does not seem to be a name. Perhaps I just add an explanatory comment?

Reply:
interesting, let's keep this then and leave o comment clarifying it's a sample AAAA DNS record